client/lib/human-date/index.js

```diff
@@ -1,28 +1,25 @@
-import i18n from 'i18n-calypso';
-import moment from 'moment';
-
+import { useTranslate } from 'i18n-calypso';
+import { useMemo, useState } from 'react';
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import { EVERY_TEN_SECONDS, useInterval } from 'calypso/lib/interval';
 const MILLIS_IN_MINUTE = 60 * 1000;
 
-export default function humanDate(
-	dateOrMoment,
-	dateFormat = 'll',
-	locale = i18n.getLocaleSlug()
-) {
-	const now = moment().locale( locale );
-	dateOrMoment = moment( dateOrMoment ).locale( locale );
+function getHumanDateString( now, date, dateFormat, moment, translate ) {
+	date = moment( date );
+	now = moment( now );
 
-	let millisAgo = now.diff( dateOrMoment );
+	let millisAgo = now.diff( date );
 	if ( millisAgo < 0 ) {
 		millisAgo = 0;
 	}
 
 	if ( millisAgo < MILLIS_IN_MINUTE ) {
-		return i18n.translate( 'just now' );
+		return translate( 'just now' );
 	}
 
 	if ( millisAgo < MILLIS_IN_MINUTE * 60 ) {
 		const minutes = Math.ceil( millisAgo / MILLIS_IN_MINUTE );
-		return i18n.translate( '%(minutes)dm ago', {
+		return translate( '%(minutes)dm ago', {
 			args: {
 				minutes: minutes,
```

Comment:
If you don't mind a last finishing touch, these objects could use property shorthands: `{ minutes }`, `{ hours }` etc.

Reply:
sure thing https://github.com/Automattic/wp-calypso/pull/57826/commits/19eb8557ead538dd3e25a2c11418b58a3d8d1245